presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -170,6 +171,8 @@ public void finish()
     {
         blocked.set(null);
         if (pageSource != null) {
+            // last chance to update operator stats to make sure lazy block scan data is recorded
+            recordPageSourceRawInputStats();
```

Comment:
I don't quite understand why this call is needed. If LazyBlocks are not loaded, the data is not read, hence no stats to update.

Reply:
This is a last chance update that shouldn't be necessary if everything else is working as expected, but attempts to harden the bytes read accounting against future refactors. I can remove it if you'd prefer, but I would note that the previous omission of accounting wouldn't have yielded incorrect results if this fallback logic in finish were there.

Reply:
@pettyjamesm I would prefer to remove this call (as long as it is not necessary).